brave/brave-core
# in-page translation

Fix https://github.com/brave/brave-browser/issues/208

This PR re-enable entries of translate service, all incurred requests will be sent to go-translate server (https://github.com/brave/go-translate) to either be relayed to MS server or be proxied through Brave to avoid introducing direct connections to google.

go-translate is now running locally at localhost:8195 manually, the URL will be updated to a brave server once go-translate is deployed by devops.

Related PR:
https://github.com/brave/go-translate/pull/1

Entry points of translation service
1) Through the translate bubble in the location bar
<img width="339" alt="Screen Shot 2019-03-19 at 3 09 27 PM" src="https://user-images.githubusercontent.com/4730197/54715536-f1373d80-4b10-11e9-982e-4506b7e94d65.png">

2) Through the context menu
<img width="178" alt="Screen Shot 2019-03-19 at 3 09 35 PM" src="https://user-images.githubusercontent.com/4730197/54645401-1cf6ec80-4a59-11e9-828e-b48f44d6018c.png">

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
1) Navigate to https://www.deutschland.de/de and click translate when the translate bubble shows up. Continue surfing through this website and pages should be translated.

2) Open a new tab to navigate to https://www.deutschland.de/de, click "Translate to English" (depends on your local language) in the context menu, current page should be translated, when click away through links in this page, they will not be translated.

## Reviewer Checklist:
